### PR TITLE
fix: prevent nil reference error when rendering project timeline events

### PR DESCRIPTION
**Pull request by ericboehs**
**Note:** The code changes here were generated via Claude Code. I ran across this issue for issues with projects in a GitHub Enterprise (ghe.com) repository. I reviewed and tested the changes myself.

## Description of Changes
Fixes a nil reference error that occurs when rendering timeline events for issues that have been added to or removed from GitHub Projects v2.

## Does this pull request fix one issue?
Yes. This PR fixes a crash that occurs when opening issues with project timeline events where the project data is nil.

## Describe how you did it
Added nil checks before accessing `item.project.title` in three locations within `lua/octo/ui/writers.lua`:

1. **`write_project_v2_item_status_changed_event()` (lines 1978-1997)**: Added nil checks in all three conditional branches before rendering project title in status change events
2. **`write_project_v2_event()` (lines 2023-2025)**: Added nil check before rendering project title in added/removed events  
3. **`write_details()` (line 568)**: Enhanced existing `vim.NIL` check to also guard against actual `nil` values

The changes are defensive - when `item.project` is nil, we simply skip displaying the project information rather than crashing.

## Describe how to verify it
1. Open an issue that has GitHub Projects v2 timeline events (added/removed/status changed)
2. If the GitHub API returns incomplete project data (project field is nil), the plugin should now render the timeline gracefully without the project title instead of crashing
3. When project data is present, behavior is unchanged

My repo's issues do automatically get added to a project via GitHub Projects Workflow (automated user for project timeline event). GitHub also added project add/remove events to their timeline last week (not sure if either of those are related).

**Error that was occurring:**
```
attempt to index field 'project' (a nil value)
stack traceback:
    ....local/share/nvim/lazy/octo.nvim/lua/octo/ui/writers.lua:2026: in function 'write_project_v2_event'
```

## Special notes for reviews
- This is a defensive fix for incomplete API data
- The change is backward compatible - existing functionality remains unchanged when project data is present
- All three functions follow the same pattern: check if `item.project` exists before accessing `item.project.title`
- No new tests needed as this handles an edge case in API responses

## Checklist
- [ ] Tests pass and code meets linting standards (based on CI, I didn't get the tests running locally)
- [x] Documentation updates not needed (bug fix only, no API changes)

### CODE REVIEW

File: lua/octo/ui/writers.lua [hunk 1]
```diff
@@ -1975,20 +1975,26 @@ function M.write_project_v2_item_status_changed_event(bufnr, item)
...
+      table.insert(vt, { item.project.title, "OctoDetailsLabel" })
+    end
   else
     table.insert(vt, { "from ", "OctoTimelineItemHeading" })
     table.insert(vt, { item.previousStatus, "OctoDetailsLabel" })
     table.insert(vt, { " to ", "OctoTimelineItemHeading" })
     table.insert(vt, { "No Status", "OctoDetailsLabel" })
-    table.insert(vt, { " in " .. conf.timeline_icons.project, "OctoTimelineItemHeading" })
-    table.insert(vt, { item.project.title, "OctoDetailsLabel" })
+    if item.project then
```
Comment (williambdean) [inline lua/octo/ui/writers.lua]: Should we render the line at all? Can you share what are in the timelineItems? Is this one specific case?

**Comment by ericboehs** [comment]
Hmm. So I found the problem! Octo.nvim is defaulting to ProjectV1 events. I found the `default_to_projects_v2 = false` in the README and flipped it to true in my config and these warnings go away. I don't remember seeing them before though. I wonder if it's related to the new [Project and project status timeline events](https://github.blog/changelog/2025-11-06-improved-onboarding-flow-for-github-projects/#🤖-api-improvements).

File: lua/octo/ui/writers.lua (hunk 1 above)
Comment (ericboehs) [reply lua/octo/ui/writers.lua]: Adding some debug logging here's what the data looked like (when I had `default_to_projects_v2 = false`):
```
[Octo Debug] ProjectV2ItemStatusChangedEvent:
  project: nil
  previousStatus: nil
  status: nil
  actor: nil
  createdAt: nil
[Octo Debug] ProjectV2ItemStatusChangedEvent:
  project: nil
  previousStatus: nil
  status: nil
  actor: nil
  createdAt: nil
[Octo Debug] ProjectV2ItemStatusChangedEvent:
  project: nil
  previousStatus: nil
  status: nil
  actor: nil
  createdAt: nil

[Octo Debug] ProjectV2ItemStatusChangedEvent:
  project: nil
  previousStatus: nil
  status: nil
  actor: nil
  createdAt: nil
[Octo Debug] ProjectV2ItemStatusChangedEvent:
  project: nil
  previousStatus: nil
  status: nil
  actor: nil
  createdAt: nil
[Octo Debug] ProjectV2ItemStatusChangedEvent:
  project: nil
  previousStatus: nil
  status: nil
  actor: nil
  createdAt: nil
[Octo Debug] ProjectV2Event (added):
  project: nil
  actor: nil
  createdAt: nil
[Octo Debug] Skipping event because project is nil
[Octo Debug] Skipping event because project is nil
[Octo Debug] Skipping event because project is nil
[Octo Debug] Skipping event because project is nil
[Octo Debug] Skipping event because project is nil
[Octo Debug] Skipping event because project is nil
[Octo Debug] Skipping added event because project is nil
```

**Comment by williambdean** [comment]
Oh, nice. So just the configuration change fixes this? Yes, this project change had caused a bit of trouble in various places as they fully deprecated in recently 

Should we close this PR and focus on documentation instead?

**Comment by ericboehs** [comment]
> So just the configuration change fixes this?

Yeah but the nil guard is still useful when accessing a v2 project.

I considered detecting project v2 and warning that a v2 project was detected but default_to_projects_v2 is set to false.

**Comment by ericboehs** [comment]
I added the warning.

File: lua/octo/ui/writers.lua [hunk 2]
```diff
@@ -11,6 +11,21 @@ local vim = vim
...
+
+--- Show a one-time warning about enabling ProjectsV2 config
+local function warn_projects_v2_config()
+  if not projects_v2_config_warned and not config.values.default_to_projects_v2 then
+    projects_v2_config_warned = true
+    vim.notify(
+      "ProjectsV2 timeline events are disabled. Enable them by setting 'default_to_projects_v2 = true' in your Octo config.",
+      vim.log.levels.WARN,
+      { title = "Octo.nvim" }
+    )
```
Comment (williambdean) [inline lua/octo/ui/writers.lua]: Can you use the notify commands built out in `utils.lua` / `notify.lua`. This will break pre-commit

Comment (ericboehs) [reply lua/octo/ui/writers.lua]: Fixed

File: lua/octo/ui/writers.lua [hunk 3]
```diff
@@ -18,11 +19,7 @@ local projects_v2_config_warned = false
 local function warn_projects_v2_config()
   if not projects_v2_config_warned and not config.values.default_to_projects_v2 then
     projects_v2_config_warned = true
-    vim.notify(
-      "ProjectsV2 timeline events are disabled. Enable them by setting 'default_to_projects_v2 = true' in your Octo config.",
-      vim.log.levels.WARN,
-      { title = "Octo.nvim" }
-    )
+    notify.warn "ProjectsV2 timeline events are disabled. Enable them by setting 'default_to_projects_v2 = true' in your Octo config."
```
Comment (ericboehs) [inline lua/octo/ui/writers.lua]: Should I just use info here?

Comment (williambdean) [reply lua/octo/ui/writers.lua]: Yeah, that is fine. Thanks for edit

Comment (ericboehs) [reply lua/octo/ui/writers.lua]: Ok. I switched to info and ditched warn.
